streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/ClientState.java

```diff
@@ -143,6 +147,14 @@ boolean reachedCapacity() {
         return assignedTaskCount() >= capacity;
     }
 
+    public Optional<Instant> followupRebalanceDeadline() {
+        return followupRebalanceDeadline;
+    }
+
+    public void setFollowupRebalanceDeadline(final Optional<Instant> followupRebalanceDeadline) {
```

Comment:
There's no reason to pass this in as an `Optional`, it should always be non-null if this method is called right?

Reply:
I'm using it here:
```
clientMetadata.state.setFollowupRebalanceDeadline(kafkaStreamsAssignment.followupRebalanceDeadline());
```
and here:
```                    
final Instant rebalanceDeadline = Instant.ofEpochMilli(time.milliseconds() + probingRebalanceIntervalMs());
rebalanceClientMetadata.state.setFollowupRebalanceDeadline(Optional.of(rebalanceDeadline));
```

Reply:
Gotcha